src/vs/workbench/services/dialogs/browser/remoteFileDialog.ts

```diff
@@ -103,8 +108,18 @@ export class RemoteFileDialog {
 		});
 	}
 
-	private getOptions(options: ISaveDialogOptions | IOpenDialogOptions): IOpenDialogOptions | undefined {
-		const defaultUri = options.defaultUri ? options.defaultUri : URI.from({ scheme: this.scheme, authority: this.remoteAuthority, path: '/' });
+	private async getOptions(options: ISaveDialogOptions | IOpenDialogOptions): Promise<IOpenDialogOptions | undefined> {
+		let defaultUri = options.defaultUri;
+		if (!defaultUri) {
+			const env = await this.remoteEnvironmentService.remoteEnvironment;
+			if (env) {
+				defaultUri = env.userHome;
+			} else if (this.scheme !== Schemas.file) {
+				defaultUri = options.defaultUri ? options.defaultUri : URI.from({ scheme: this.scheme, authority: this.remoteAuthority, path: '/' });
```

Comment:
This branch doesn't really make a lot of sense to me. `this.remoteAuthority` is actually `this.windowService.getConfiguration().remoteAuthority` (from the ctor), and that has a direct correlation with a defined remote environment. i.e. I think the case when the remote environment would be null is exactly the case when `remoteAuthority` is null/undefined.

Please also test with `eamodio.remotehub` which will trigger this case because of their URIs where the scheme will not be `file` and will have `this.remoteAuthority` null/undefined. The constructed URI will lack an `authority` component.

Reply:
Good background, thank!